File: Python/6py 2/num_primorial.py
"""
|--------------------------------------------------------------------------
| Primorial Of a Number
|--------------------------------------------------------------------------
|
"""

def is_prime(num):
   return num > 1 and all(num % i for i in range(2, int(num**0.5) + 1))
def num_primorial(n):
   primes, num = [], 2
   while len(primes) < n:
      if is_prime(num): primes.append(num)
      num += 1
   res = 1
   for prime in primes: res *= prime
   return res


def num_primorial(n):
   primorial, x, n = 2, 3, n-1
   while n:
      if all(x % d for d in range(3, int(x ** .5) + 1, 2)):
         primorial *= x
         n -= 1
      x += 2
   return primorial


from gmpy2 import next_prime
def num_primorial(n: int) -> int:
   p = prev = 1
   for _ in range(n):
      prev = next_prime(prev)
      p *= prev
   return p


print(num_primorial(3),30)
print(num_primorial(4),210)
print(num_primorial(5),2310)
print(num_primorial(8),9699690)
print(num_primorial(9),223092870)

File: PHP/6php 2/numPrimorial.php
<?php
/*
|--------------------------------------------------------------------------
| Primorial Of a Number
|--------------------------------------------------------------------------
|
*/

function isPrime($num) {
   if ($num <= 1) return false;
   for ($i = 2; $i <= sqrt($num); $i++) {
      if ($num % $i == 0) return false;
   }
   return true;
}
function numPrimorial($n) {
   $res = [];
   $num = 2;
   while (count($res) < $n) {
      if (isPrime($num)) array_push($res, $num);
      $num++;
   }
   return array_product($res);
}


function numPrimorial($n){
   $p = 0;
   while($n != 0){
      $p++;
      if(!preg_match('/^1?$|^(11+?)\1+$/x', str_repeat('1', $p))){
         $res[] = $p;
         $n--;
      }
   }
   return array_product($res);
}


function numPrimorial($n) {
   $p_num = $res = 1;
   for ($c = 0; $c <= $n; $p_num++) {
      $fg = 0;
      for ($i = 1; $i <= $p_num; $i++) {
         if (($p_num % $i) == 0) $fg++;
      }
      if ($fg < 3) {
         $res *= $p_num;
         $c++;
      }
   }
   return $res;
}


print_r(numPrimorial(3) . ' ' . 30 . '
');
print_r(numPrimorial(4) . ' ' . 210 . '
');
print_r(numPrimorial(5) . ' ' . 2310 . '
');
print_r(numPrimorial(8) . ' ' . 9699690 . '
');
print_r(numPrimorial(9) . ' ' . 223092870 . '
');